docs/root/configuration/http/http_filters/stateful_session_filter.rst

```diff
@@ -0,0 +1,80 @@
+.. _config_http_filters_stateful_session:
+
+Stateful session
+================
+
+Stateful session is an HTTP filter which sets an override host based on the extensible session state
+and updates the session state based on the final selected upstream host. The override host will
+eventually overwrites the load balancing result. This filter can implement session sticky without using
+a hash-based load balancer.
+And by extending the session state, this filter also allows more flexible control over the results of
+the load balancing.
+
+
+Overview
+--------
+
+Session sticky is an important feature of LB. Requests belonging to the same session should be guaranteed
+to be routed to the same backend host.
+HTTP session sticky of Envoy is generally achieved through hash-based load balancer. And the filter is a
+supplement. We should use the filter in the following cases:
+
+* The case where more stable session sticky is required. For example, when a host is marked as degrade and
+  it is hoped that the requests for existing sessions will not be affected. At this time, stateful session
+  filter is needed.
+* The case where a non hash-based load balancer (Random, Round Robin, etc.) is used and session sticky
+  is still required. If stateful session is enabled in this case, new requests will be routed to the
+  corresponding upstream host based on the result of load balancing. The requests belonging to specific
+  session will be routed to specific host.
+
+
+Configuration
+-------------
+
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.filters.http.stateful_session.v3.StatefulSession>`
+* This filter should be configured with the name *envoy.filters.http.stateful_session*.
+
+How it works
+------------
+
+Stateful session configuration consists of two parts. An :ref:`extensible session state
```

Comment:
Nit: can you format this as a proper RST llist and validate it looks right?

Reply:
get it.